src/postprocessing.jl

```diff
@@ -0,0 +1,550 @@
+## Postprocessing
+
+function postprocess!(
+    color::AbstractVector{<:Integer},
+    star_or_tree_set::Union{StarSet,TreeSet},
```

Comment:
The splitting of the code into 4 functions makes it difficult to review and maintain. I think we can get away with only two, separating `StarSet` vs `TreeSet` but keeping a unified logic for symmetric and bidirectional. We can fill a `column_color_used` and `row_color_used` in both cases, and then recover `color_used` with a `max` operation. That would make things much clearer

Reply:
It is what I asked you Guillaume during our last meeting and you said yes and even suggested to split again for bicoloring.

Reply:
I will do my best to fuse the functions such we can keep only 2.

Reply:
> It is what I asked you Guillaume during our last meeting and you said yes and even suggested to split again for bicoloring.

I thought we would only get one split, now two, sorry for the misunderstanding